crates/wast/src/core/resolve/names.rs

```diff
@@ -610,8 +635,12 @@ impl<'a, 'b> ExprResolver<'a, 'b> {
             }
 
             StructSet(s) | StructGet(s) | StructGetS(s) | StructGetU(s) => {
-                self.resolver.resolve(&mut s.r#struct, Ns::Type)?;
-                self.resolver.fields.resolve(&mut s.field, "field")?;
+                let type_index = self.resolver.resolve(&mut s.r#struct, Ns::Type)?;
+                self.resolver
+                    .fields
+                    .get(&type_index)
+                    .unwrap()
```

Comment:
This should not be an `unwrap()` but instead should return some form of error message. Note that it should not always be required to be `Some`, however, because the `field` may not actually need resolution. Only if the field is an `Id` is it required that this `.get(...)` succeeds in which case an error would be emitted. For example `struct.get 1000 1000` should be ok (even if it's invalid), but `struct.get 1000 $foo` would only be valid if type `1000` is defined.

Reply:
Fixed.